flepimop/gempyor_pkg/src/gempyor/shared_cli.py

```diff
@@ -0,0 +1,186 @@
+        show_default=True,
+        help="the parallelization factor",
+    ),
+    click.option(
+        "-n",
+        "--nslots",
+        envvar="FLEPI_NUM_SLOTS",
+        type=click.IntRange(min=1),
+        help="override the # of simulation runs in the config file",
+    ),
+    click.option(
+        "--in-id",
+        "in_run_id",
+        envvar="FLEPI_RUN_INDEX",
+        type=str,
+        show_default=True,
+        help="Unique identifier for the run",
+    ),
+    click.option(
+        "--out-id",
+        "out_run_id",
+        envvar="FLEPI_RUN_INDEX",
+        type=str,
+        show_default=True,
+        help="Unique identifier for the run",
+    ),
+    click.option(
+        "--in-prefix",
+        "in_prefix",
+        envvar="FLEPI_PREFIX",
+        type=str,
+        default=None,
+        show_default=True,
+        help="unique identifier for the run",
+    ),
+    click.option(
+        "-i",
+        "--first_sim_index",
+        envvar="FIRST_SIM_INDEX",
+        type=click.IntRange(min=1),
+        default=1,
+        show_default=True,
+        help="The index of the first simulation",
+    ),
+    click.option(
+        "--stochastic/--non-stochastic",
+        "stoch_traj_flag",
+        envvar="FLEPI_STOCHASTIC_RUN",
+        default=False,
+        help="Run stochastic simulations?",
+    ),
+    click.option(
+        "-s",
+        "--seir_modifiers_scenarios",
+        envvar="FLEPI_SEIR_SCENARIO",
+        type=str,
+        default=[],
+        multiple=True,
+        help="override/select the transmission scenario(s) to run",
+    ),
+    click.option(
+        "-d",
+        "--outcome_modifiers_scenarios",
+        envvar="FLEPI_OUTCOME_SCENARIO",
+        type=str,
+        default=[],
+        multiple=True,
+        help="override/select the outcome scenario(s) to run",
+    ),
+    click.option(
+        "-c",
+        "--config",
+        "config_filepath",
+        envvar="CONFIG_PATH",
+        type=click.Path(exists=True),
+        required=False,  # deprecated = ["-c", "--config"],
+        # preferred = "CONFIG_FILES...",
+        help="Deprecated: configuration file for this simulation",
+    ),
+]
+""" List of `click` options that will be applied by `option_config_files` """
+
+
+def option_config_files(function: callable) -> callable:
+    """`click` Option decorator to apply handlers for all options"""
+    reduce(lambda f, option: option(f), config_file_options, function)
+    return function
+
+
+def parse_config_files(
+    config_files: list[str],
+    config_filepath: str,
+    in_run_id: str,
+    out_run_id: str,
+    seir_modifiers_scenarios: list[str],
+    outcome_modifiers_scenarios: list[str],
+    in_prefix: str,
+    nslots: int,
+    jobs: int,
+    write_csv: bool,
+    write_parquet: bool,
+    first_sim_index: int,
+    stoch_traj_flag: bool,
+) -> None:
+    """Parse the configuration file(s) and override with command line arguments"""
```

Comment:
Documentation?

Reply:
Hmm, what is the right way to avoid double work here? The click options are in my mind the right place to have the documentation, and this def is *not* intended to public interface. I don't want us to get into double maintaining / inconsistency between the allowed options / arguments to this.

Reply:
along those lines: I propose a separate issue/PR here for a bit more focused refactor here to make options and such be a bit more dynamic. Swap to a kwargs-y approach, make the options list defined earlier in the file be named and use those names in the loops.

Does that seem reasonable?

Reply:
I think the way to avoid double work here is to put the documentation with this function. That's fine that they're not intended for the public interface (we should start [denoting internal with a leading underscore](https://peps.python.org/pep-0008/#descriptive-naming-styles) if we're going to take care to define an external vs internal interface at this stage), but that doesn't exclude it from requiring documentation. We still need to know how to use the function internally when building out future CLI interfaces. 

I'm not sure I follow the second comment that well. Why would we move to kwargs for `parse_config_files` if we're providing them every time (at least from what I see in current usage) anyways, do we anticipate only providing a few of these args?

Reply:
so if `parse_config_files` is kwargs-driven, then effectively the arguments can be driven by the names set in the arguments/options objects. Then whatever we do with those options in the future (and maybe that other people do with options for plugins), the parse function "just works". that clearer?

Reply:
Oh, you mean using `**kwargs`. Sure, that could be done now or in a future PR doesn't matter too much either way, but it doesn't alleviate the need for documentation. One downside of `**kwargs` is how to construct a valid call to this function becomes a bit more ambiguous since the valid arguments are not enumerated in the function definition itself. That could be addressed by well maintained documentation though.